c10/core/TensorImpl.h

```diff
@@ -573,7 +587,24 @@ struct C10_API TensorImpl : public c10::intrusive_ptr_target {
       const caffe2::TypeMeta data_type,
       c10::optional<c10::Device>);
 
+  // Private constructor for lazily copying/viewing tensors.
```

Comment:
I think you should tag this constructor with the VIEW tag. In particular, it's non-obvious that this constructor DOESN'T copy sizes/strides. (This is the desired behavior, but one could easily be deceived by the fact that this takes a Tensor)